frontend/src/components/Analytics.tsx

```diff
@@ -105,38 +91,31 @@ const Analytics: React.FC = () => {
       .catch((err) => console.log(err));
   }, []);
 
+  // Fetch user attempted questions id, title and difficulty
   useEffect(() => {
     if (allQuestionIds.length > 0) {
-      const requestBody = {
-        ids: allQuestionIds,
-      };
-      axios
-        .post(QUESTION_HOST + "/questionbyid", requestBody, {
-          headers: authHeader(),
-        })
+      fetchAttemptedQuestions(allQuestionIds)
         .then((response) => {
           const data = response.data;
-          const titleAndDifficulty = data.map((item: any) => ({
+          const idAndtitleAndDifficulty = data.map((item: any) => ({
```

Comment:
Perhaps camelCase would be good here :)

```suggestion
          const idAndTitleAndDifficulty = data.map((item: any) => ({
```